### PR TITLE
Metamist Comments Frontend

### PR DESCRIPTION
**Pull request by dancoates**

### CODE REVIEW

File: web/src/pages/analysis/AnalysisGrid.tsx [hunk 1]
```diff
@@ -58,14 +50,7 @@ export const AnalysisGrid: React.FC<{
...
                             <SUITable.Cell>
-                                <AnalysisLink
-                                    id={a.id}
-                                    onClick={(e) => {
-                                        e.preventDefault()
-                                        e.stopPropagation()
-                                        setAnalysisIdToView(a.id)
-                                    }}
-                                />
+                                <AnalysisLink id={a.id} />
```
Comment (ckrishk) [inline web/src/pages/analysis/AnalysisGrid.tsx:53]: Nice!

File: web/src/pages/comments/CommentAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import RobotIcon from '@mui/icons-material/SmartToy'
+import { Avatar, SxProps } from '@mui/material'
+
+// Generate a random unique colour from the users username
+const colorFromStr = (str: string) => {
+    let hash = 0
+    if (str.length === 0) return `hsl(0 0 50)`
```
Comment (ckrishk) [inline web/src/pages/comments/CommentAvatar.tsx:7]: does this need to be a component ?

File: web/src/pages/comments/commentConfig.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        case 'GraphQLParticipant':
+            return `participant:${comment.entity.participantId}`
+
+        case 'GraphQLProject':
+            return `project:${comment.entity.projectId}`
+
+        case 'GraphQLSample':
+            return `sample:${comment.entity.sampleId}`
+        case 'GraphQLSequencingGroup':
```
Comment (ckrishk) [inline web/src/pages/comments/commentConfig.ts:64]: nit: formatting - missing new line

**Review by ckrishk — APPROVED** [review]
Great work with this Dan! Looking good to me .

File: web/src/index.tsx [hunk 4]
```diff
@@ -16,7 +16,21 @@ const httpLink = createHttpLink({
 
 const client = new ApolloClient({
     link: httpLink,
-    cache: new InMemoryCache(),
+    cache: new InMemoryCache({
+        possibleTypes: {
+            // Not sure why apollo can't get this from the schema,
+            // but this is required so that fields on the union
+            // types are retrievable.
```
[the remark this answers is not in the document]
Comment (dancoates) [reply web/src/index.tsx]: I've added an additional note to the comment about how we could automatically generate these in the future if the manual specification gets too unwieldy

File: web/src/pages/comments/commentConfig.ts (hunk 3 above)
Comment (dancoates) [reply web/src/pages/comments/commentConfig.ts:65]: Thanks! I've fixed this up

File: web/src/App.tsx [hunk 5]
```diff
@@ -1,21 +1,38 @@
+import { Alert, Snackbar } from '@mui/material'
 import * as React from 'react'
 import { BrowserRouter as Router } from 'react-router-dom'
+import Routes from './Routes'
+import NavBar from './shared/components/Header/NavBar'
+import { ViewerContext, useViewer } from './viewer'
 
 // this wasn't working, so added import to HTML
 // import 'bootstrap/dist/css/bootstrap.min.css'
```
[the remark this answers is not in the document]
Comment (dancoates) [reply web/src/App.tsx]: I agree, have removed

**Review by milo-hyben — APPROVED** [review]
Hey @dancoates, excellent work, I really like how you had split all the comment's code into smaller components, it is easy to follow. That PaddedPage is a nice touch as well!
I have not tested GUI functionality, only view the code. I could not see any obvious issues.

**Pull request merged**